### ISSUE TITLE
GameData\000_USITools\Comms.cfg still misses @

### ISSUE THREAD

**Issue by Gordon-Dry** [UmbraSpaceIndustries/USITools#131]
Line 1:
`@PART[*]:HAS[!MODULE[ModuleDataTransmitter],MODULE[ModuleCommand]]`
should be
`@PART[*]:HAS[!MODULE[ModuleDataTransmitter],@MODULE[ModuleCommand]]`

**Comment by 7ranceaddic7** [comment]
Actually, I think they have to be split up; you can't edit (@) and bang (!) within the same HAS directive ... I think.

@PART[*]:HAS[!MODULE[ModuleDataTransmitter]]:HAS[@MODULE[ModuleCommand]]

**Comment by Gordon-Dry** [comment]
I will test that

**Comment by Gordon-Dry** [comment]
Looks good

**Comment by 7ranceaddic7** [comment]
Just so you know, I corrected another config in the opposite direction.

Having separate :HAS directives makes MM perform and execute the processed PART TWICE.  So with ...

@PART[*]:HAS[MODULE[someModule]]:HAS[MODULE[anotherModule]]
{
    +module=whatever
}

If PART has both someModule & anotherModule, whatever will be created at each instance of *Module.

But with ...

@PART[*]:HAS[MODULE[someModule,anotherModule]]
{
    +module=whatever
}

Any PART with both someModule & anotherModule will have whatever created only once.

**Comment by Gordon-Dry** [comment]
ofc - dunno why I overlooked that one ...

**Comment by Gordon-Dry** [comment]
My first one was right btw - I have seen it often and use it myself.

You misleaded me... ;)

So,
`@PART[*]:HAS[!MODULE[ModuleDataTransmitter],@MODULE[ModuleCommand]]`

**Comment by 7ranceaddic7** [comment]
Yeah, looking back that would make more sense given what I said about someModule and anotherModule.
